src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,173 @@
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    public Command getSetTargetAngleCommand(Rotation2d targetAngle) {
+        return new FunctionalCommand(
+                () -> generateAngleMotorProfile(targetAngle),
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    public Command getSetTargetElevatorPositionCommand(double targetElevatorPosition) {
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(targetElevatorPosition),
+                this::setTargetElevatorPositionFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            stopAngleMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        setAngleMotorsVoltage(targetState);
+    }
+
+    private void setTargetElevatorPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            stopElevatorMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        setElevatorMotorsVoltage(targetState);
+    }
+
+    private void setAngleMotorsVoltage(TrapezoidProfile.State targetState){
+        masterAngleMotor.setVoltage(calculateAngleMotorOutput(targetState));
+        followerAngleMotor.setVoltage(calculateAngleMotorOutput(targetState));
+    }
+
+    private void setElevatorMotorsVoltage(TrapezoidProfile.State targetState){
+        masterElevatorMotor.setVoltage(calculateElevatorMotorOutput(targetState));
+        followerElevatorMotor.setVoltage(calculateElevatorMotorOutput(targetState));
+    }
+
+    private double calculateAngleMotorOutput(TrapezoidProfile.State targetState){
+        double pidOutput = ArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAngleMotorPosition().getDegrees(),
+                targetState.position
+        );
+        double feedForward = ArmConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                Units.degreesToRadians(targetState.velocity)
+        );
+
+        return pidOutput + feedForward;
+    }
+
+    private double calculateElevatorMotorOutput(TrapezoidProfile.State targetState){
+        double pidOutput = ArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                getElevatorMotorPositionRevolutions(),
+                targetState.position
+        );
+        double feedForward = ArmConstants.ELEVATOR_FEEDFORWARD.calculate(
+                Units.rotationsToRadians(targetState.position),
+                Units.rotationsToRadians(targetState.velocity)
```

Comment:
Why does this need to be radians? Check also which params this function can accept if you're putting the right ones

Reply:
and it needs to be radians because it has to do cos or smth like that. ur the one who told us to convert it to radians

Reply:
also, I checked the params. it said it takes velocity and acceleration. how do I get the accelaration?

Reply:
The cos is for an arm, you're using an elevator. 
You put position for the velocity and velocity for the accel. You have an option to only put velocity, use that option.